src/libsyntax/attr.rs

```diff
@@ -312,6 +312,15 @@ pub fn find_export_name_attr(diag: &SpanHandler, attrs: &[Attribute]) -> Option<
     })
 }
 
+pub fn contains_extern_indicator(attrs: &[Attribute]) -> bool {
+    contains_name(attrs, "no_mangle") ||
+        contains_name(attrs, "export_name") ||
+        first_attr_value_str_by_name(attrs, "linkage").map(|v| match &*v {
```

Comment:
I think it's fine to not check for this as even if the linkage is external it still doesn't imply that the function itself is not mangled (e.g. linkage is orthogonal to mangling)